src/languages.ts

```diff
@@ -130,24 +130,29 @@ export const languages: {
         close: '"'
       },
       {
-        open: "'",
-        close: "'"
+        open: "#=",
+        close: "=#"
       }
     ],
     inlineOpenTokens: [],
     openTokens: [
       "if",
       "struct",
+      "abstract type",
+      "primitive type",
       "begin",
       "let",
       "for",
```

Comment:
@singularitti Also, if non-generator-like `for` requires `do` or another token, you could change this to:
```suggestion
      "for(?=.*do)",
```

This should now only match expressions like: `for ... do`

Reply:
Hi, I am trying to use this pattern `"^\s*for"` (A `for` with only blank characters before it) to replace `"for"`, but it does not seem to succeed. The below generator expression
```julia
[getfield(x, :a) for x in object.b.c],
```
is still treated as a `for` loop.

Reply:
![for](https://user-images.githubusercontent.com/25192197/64091747-ae180200-cd1f-11e9-8c94-0a8e28e890b0.png)
As an example, I only want the first `for` be matched.

Reply:
I'm not familiar with Julia. Why the `for` preceeded by @softscope shouldn't be matched? This might bring problems with the matching `end` for it.

Check if one of these tokens work well enough for `for`: 

- `for(?!.*\]))`
- `(?<!\[)for`

Reply:
Hi @polvalente, thank you for your reply. I am not an expert in regex, so I just wrote the simplest example. Yes, indeed, that will break the highlighting.

I cannot figure out a good regex to distinguish the `for` loops and generator expressions. Because Julia is so flexible that it allows a lot of ways to write a `for` loop. The only difference between them just seems to be `for` loops have an `end`. I will list some test examples below. I hope you can figure out a working way to distinguish them. Thank you!

Below are normal `for` loops so should be paired with `end`.
```julia
# Those should be matched.
for i = 1:10
           s += i
       end

@softscope for i in 1:10
           s += i
       end

softscope(Main, :(for i = 1:10
           s += i
       end))

for i in 1:10; println(i); end

for i in [1, 2, 3]
	println(i)
end

for i in collect(1:10)
	println(i)
end
```

Below are generator expressions and are not paired with `end`:
```julia
# Those should not be matched.
sum(1/n^2 for n=1:1000)

 map(tuple, (1/(i+j) for i=1:2, j=1:2), [1 3; 2 4])

[(i,j) for i=1:3 for j=1:i if i+j == 4]
```

Reply:
Assuming you already have a syntax highlighter, the simplest solution is to add `()` and `[]` as ignore-blocks. That way, the normal syntax highlighter should take care of the non-matching `for`s and `if`s.

Reply:
I have tried this method. It seems that with the following test text:
```julia
sum(1/n^2 for n=1:1000)

map(tuple, (1/(i+j) for i=1:2, j=1:2), [1 3; 2 4])

[(i,j) for i=1:3 for j=1:i if i+j == 4]

for i = 1:10
    s += i
end

@softscope for i in 1:10
    s += i
end

softscope(Main, :(for i = 1:10
    s += i
end))

for i in 1:10; println(i); end

for i in [1, 2, 3]
println(i)
end

for i in collect(1:10)
println(i)
end
```

![image](https://user-images.githubusercontent.com/25192197/64143653-fe09ce00-cdde-11e9-8548-4faba01a0308.png)

The `map(tuple, (1/(i+j) for i=1:2, j=1:2), [1 3; 2 4])` is not working because there is one `)` before the `for`.

Reply:
Oh wow. I think there is a problem with how I implemented ignore blocks. I'll refactor it in the coming weekend.

Basically, I didn't think to support nesting. As such, `((x) for)` will be passed to the coloring function as `     for)` (because the first closing paren will close the first open paren).

Luckily, I already have an idea on how to solve it (by using stacks).

Reply:
https://github.com/noplay/vscode-rainbow-end/issues/18

Reply:
Nevermind the weekend: https://github.com/noplay/vscode-rainbow-end/pull/19
HAHA

Reply:
Cool! Cannot wait to have a try!

Reply:
> I have tried this method. It seems that with the following test text:
> 
> ```julia
> sum(1/n^2 for n=1:1000)
> 
> map(tuple, (1/(i+j) for i=1:2, j=1:2), [1 3; 2 4])
> 
> [(i,j) for i=1:3 for j=1:i if i+j == 4]
> 
> for i = 1:10
>     s += i
> end
> 
> @softscope for i in 1:10
>     s += i
> end
> 
> softscope(Main, :(for i = 1:10
>     s += i
> end))
> 
> for i in 1:10; println(i); end
> 
> for i in [1, 2, 3]
> println(i)
> end
> 
> for i in collect(1:10)
> println(i)
> end
> ```
> 
> ![image](https://user-images.githubusercontent.com/25192197/64143653-fe09ce00-cdde-11e9-8548-4faba01a0308.png)
> 
> The `map(tuple, (1/(i+j) for i=1:2, j=1:2), [1 3; 2 4])` is not working because there is one `)` before the `for`.

This test now works:
![work](https://user-images.githubusercontent.com/25192197/64211455-390a1100-ce74-11e9-9c2a-aa6156640b31.png)